tools/topology2/include/common/connection.conf

```diff
@@ -0,0 +1,44 @@
+# Class definition for DAPM graph connection objects
+# These are instantiated as follows:
+# Example:
+#	Object.connection."N" {
+#		source	"dai.SSP.1.capture"
+#		sink	"Object.volume-playback.0.endpoint.0"
+#	}
+# The above object specifies a DAPM graph route: "dai.SSP.1.capture -> buffer.2.1"
+# where buffer.2.1 is the widget that endpoint 0 refers to in volume-playback pipeline
+# and N is the unique instance number for the connection object within the same alsaconf node.
+#
+# Example: endpoint connection
+#	Object.connection."N" {
+#		source	"Object.buffer.2"
+#		sink	"Object.pga.2"
+#	}
+# The above object specifies the route: "buffer.2 -> pga.2" where buffer.2 is the buffer object
+# with index 2 and pga.2 refers to the PGA object with index 2 in the parent object.
+# and N is the unique instance number for the connection object within the same alsaconf node.
+#
+#
+Class.Custom."connection" {
```

Comment:
I added the custom group for all class that need special handling in the alsatplg compiler. Like the connection class because it needs to "decode" the source/sink widget names